main.go

```diff
@@ -275,40 +275,26 @@ func createFailureIssue(ctx upgrade.Context, repoOrg string, repoName string) (s
 	}
 	title := fmt.Sprintf("Workflow failure: Upgrade provider | %v %v, %v @ %v:%v", mth, day, yr, hr, minStr)
 
-	getLatestWorkflowRun := exec.CommandContext(ctx, "gh", "run", "list",
-		"--workflow="+"upgrade-provider.yml",
-		"--repo="+repoOrg+"/"+repoName,
-		"--limit=1",
-		"--json=url",
-	)
-	ghRunBytes := new(bytes.Buffer)
-	getLatestWorkflowRun.Stdout = ghRunBytes
-	err := getLatestWorkflowRun.Run()
-	if err != nil {
-		return "createFailureIssue error: failed to query latest workflow run", err
-	}
-	var workflowRunUrls []struct {
-		Url string `json:"url"`
-	}
-	err = json.Unmarshal(ghRunBytes.Bytes(), &workflowRunUrls)
-	if err != nil {
-		return "createFailureIssue error: failed to unmarshal `gh run list` output", err
-	}
-	if len(workflowRunUrls) == 0 {
-		return "createFailureIssue error: no workflow runs detected", err
-	}
-	workflowUrl := workflowRunUrls[0].Url
+	// From https://docs.github.com/en/actions/learn-github-actions/variables:
+	//
+	//	If you need to use a workflow run's URL from within a job, you can combine
+	//	these variables:
+	//	$GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID
+	workflowURL := fmt.Sprintf("%s/%s/actions/runs/%s",
+		os.Getenv("GITHUB_SERVER_URL"),
+		os.Getenv("GITHUB_REPOSITORY"),
+		os.Getenv("GITHUB_RUN_ID"))
 
 	getIssues := exec.CommandContext(ctx, "gh", "search", "issues",
 		"Workflow failure: Upgrade provider",
 		"--repo="+repoOrg+"/"+repoName,
 		"--json=title,number",
 		"--state=open",
-		"--author=pulumi-bot",
+		"--author=@me",
```

Comment:
When CI is run locally, it always creates a new issue. This is categorically fixed by not hard-coding `pulumi-bot` and instead referencing the current user.

This will also improve the experience for any user who isn't Pulumi.